internal/controllers/machineset/machineset_controller.go

```diff
@@ -313,6 +313,10 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 
 	// Always updates status as machines come up or die.
 	if err := r.updateStatus(ctx, cluster, machineSet, filteredMachines); err != nil {
+		if errors.Is(err, remote.ErrClusterLocked) {
+			log.V(5).Info("Requeuing because another worker has the lock on the ClusterCacheTracker")
```

Comment:
Let's add an additional log before l.317 if syncErr != nil. Because otherwise it wouldn't show up anywhere if we hit this case

Reply:
This `if` block is not needed anymore after rebased onto main branch, because the handling of ErrClusterLocked is already in the reconcile func and not needed here.

The [reconcile func ](https://github.com/smartxworks/cluster-api/blob/main/internal/controllers/machineset/machineset_controller.go#L181-L190) does not print the aggregated error detail when the error contains ErrClusterLocked. If we should print the errors, I can add it as well.

Reply:
I'll take another look

Reply:
Now it's fine. I think in the previous variant the syncErr wouldn't have been printed. Now it's fine as the error we propagate up is eventuallyl printed by controller-runtime

Reply:
Hm it's a bit annoying but I would say let's add a log if the err is the ErrClusterLocked and syncErr != nil (before l.316)

With a godoc comment that we have to do it because otherwise the parent func doesn't print the error and it get's lost

Reply:
The [reconcile func ](https://github.com/smartxworks/cluster-api/blob/main/internal/controllers/machineset/machineset_controller.go#L181-L190)does not print the aggregated error detail when the error contains ErrClusterLocked.  I think we should print the errors there so as to cover all cases instead of here for only one case.

Reply:
Is there any way we can check that it contains more than one error? I'm asking because if we also log every time only the ErrClusterLocked occurs it will be very very spamy (that's the reason why the log there is on level 5)

Reply:
Sure, updated in latest commits.